docs/CustomApp.md

```diff
@@ -3,110 +3,244 @@ layout: default
+            ),
+            typeof window !== 'undefined' && window.devToolsExtension
+                ? window.devToolsExtension()
+                : f => f
+            // add your own enhancers here
+        )
+    );
+    sagaMiddleware.run(saga);
+    return store;
+};
+```
+
+You can use this script as a base and then add your own middleares or enhancers, e.g. to allow store persistence with [redux-persist](https://github.com/rt2zz/redux-persist).
+
+Then, use the `<Admin>` component as you would in a standalone application. Here is an example with 3 resources: `posts`, `comments`, and `users`
+
+```js
+// in src/App.js
+import React from 'react';
+import { Provider } from 'react-redux';
+import createHistory from 'history/createHashHistory';
+import { Admin, Resource } from 'react-admin';
 import restProvider from 'ra-data-simple-rest';
 import defaultMessages from 'ra-language-english';
 
+import createAdminStore from './createAdminStore';
+import messages from './i18n';
+
 // your app components
 import Dashboard from './Dashboard';
 import { PostList, PostCreate, PostEdit, PostShow } from './Post';
 import { CommentList, CommentEdit, CommentCreate } from './Comment';
 import { UserList, UserEdit, UserCreate } from './User';
-// your app labels
-import messages from './i18n';
 
-// create a Redux app
-const reducer = combineReducers({
-    admin: adminReducer,
-    i18n: i18nReducer('en', messages['en']),
-    form: formReducer,
-    routing: routerReducer,
-});
-const sagaMiddleware = createSagaMiddleware();
+// side effects
+const authProvider = () => Promise.resolve();
+const dataProvider = restProvider('http://path.to.my.api/');
+const i18nProvider = locale => {
+    if (locale !== 'en') {
+        return messages[locale];
+    }
+    return defaultMessages;
+};
 const history = createHistory();
-const store = createStore(reducer, undefined, compose(
-    applyMiddleware(sagaMiddleware, routerMiddleware(history)),
-    window.devToolsExtension ? window.devToolsExtension() : f => f,
-));
+
+const App = () => (
+    <Provider
+        store={createAdminStore({
+            authProvider,
+            dataProvider,
+            i18nProvider,
+            history,
+        })}
+    >
+        <Admin
+            authProvider={authProvider}
+            history={history}
+            title="My Admin"
+        >
+            <Resource name="posts" list={PostList} create={PostCreate} edit={PostEdit} show={PostShow} />
+            <Resource name="comments" list={CommentList} edit={CommentEdit} create={CommentCreate} />
+            <Resource name="users" list={UserList} edit={UserEdit} create={UserCreate} />
+        </Admin>
+    </Provider>
+);
+
+export default App;
+```
+
+**Tip**: One thing to pay attention to is that you must pass the same `history` and `authProvider` to both the redux Store creator and the `<Admin>` component. But you don't need to pass the `dataProvider` or the `i18nProvider`.
```

Comment:
This rings a bell. The `authProvider` is passed both to the saga and to the React context. We're better with the `<TranslationProvider>`, which delegates all side effects to the related saga. I think the way we access the `authProvider` should be refactored.

Reply:
Possible alternative: Putting an `<AuthProvider>` component at the top, which puts the `authProvider` in the context (actually done in `<Admin>`), then using the `getContext` effect in the `auth` saga. Allows the store to be initialized without explicitely passing the `authProvider`:

```diff
const App = () => (
+<AuthProvider source={authProvider}>
    <Provider
        store={createAdminStore({
-           authProvider,
            dataProvider,
            i18nProvider,
            history,
        })}
    >
        <Admin
-           authProvider={authProvider}
            history={history}
            title="My Admin"
        >
            <Resource name="posts" list={PostList} create={PostCreate} edit={PostEdit} show={PostShow} />
            <Resource name="comments" list={CommentList} edit={CommentEdit} create={CommentCreate} />
            <Resource name="users" list={UserList} edit={UserEdit} create={UserCreate} />
        </Admin>
    </Provider>
+</AuthProvider>
);
```

Reply:
Forget what I just wrote: saga's `getContext` does not read in the React context, but in the saga context - these are two separate things.